tests/unit/s2n_ecc_evp_test.c

```diff
@@ -345,5 +360,119 @@ int main(int argc, char **argv) {
         }
     }
 
+    /* Test that the client does not negotiate a group that was not
+     * offered in EC preferences */
+    {
+        const struct s2n_security_policy* security_policy = NULL;
+        DEFER_CLEANUP(struct s2n_connection* conn = s2n_connection_new(S2N_CLIENT),
+        s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(conn);
+        EXPECT_SUCCESS(s2n_connection_set_cipher_preferences(conn, "20190802"));
+        EXPECT_SUCCESS(s2n_connection_get_security_policy(conn, &security_policy));
+
+        /* Ensure that the applied security policy uses the correct ecc preferences */
+        const uint8_t EXPECTED_ECC_PREF_COUNT = 2;
+        EXPECT_EQUAL(security_policy->ecc_preferences, &s2n_ecc_preferences_20140601);
+        EXPECT_EQUAL(security_policy->ecc_preferences->count, EXPECTED_ECC_PREF_COUNT);
+
+        /* Setup & verify invalid curves, which will be selected by a malicious server */
+        const struct s2n_ecc_named_curve* const unrequested_curves[] = {
+                &s2n_unsupported_curve,
+                &s2n_ecc_curve_secp521r1,
+        };
+
+        /* Test that client's preferred ecc's do NOT include the curves selected by the server */
+        for (size_t ix = 0; ix < EXPECTED_ECC_PREF_COUNT; ix++) {
+            const struct s2n_ecc_named_curve* curve = security_policy->ecc_preferences->ecc_curves[ix];
+            for (uint8_t iy = 0; iy < s2n_array_len(unrequested_curves); iy++) {
+                EXPECT_NOT_EQUAL(curve->name, unrequested_curves[iy]->name);
+            }
+        }
+
+        /* Verify that the client broadcasts an error code when the server attempts to
+         * negotiate a curve that was never offered */
+        for (size_t i = 0; i < s2n_array_len(unrequested_curves); i++) {
+            struct s2n_ecc_evp_params server_params = {0}, client_params = {0};
```

Comment:
We don't declare multiple variables per line for readability reasons.
```suggestion
            struct s2n_ecc_evp_params server_params = { 0 };
            struct s2n_ecc_evp_params client_params = { 0 };
```

Reply:
This was already done in the tests: https://github.com/aws/s2n-tls/blob/dffd2aa495549249018ee9334a4a4c661d49eb08/tests/unit/s2n_ecc_evp_test.c#L311. Also, I don't see this as reducing readability if there are only two variables declared